connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -654,6 +655,7 @@ private void closePartitions(Collection<TopicPartition> topicPartitions, boolean
             topicPartitions.forEach(currentOffsets::remove);
         }
         updatePartitionCount();
+        topicPartitions.forEach(lastCommittedOffsets::remove);
```

Comment:
Is `removeAll` a supported operation by the underlying set of the hashmap here? Is it applicable?
(same question above for `currentOffsets`)

Reply:
Ah, neat trick. Ran unit+integration tests locally after making this change and they all passed; looks like the operation is supported.